### PR TITLE
Companion PR for `Remove the service builder`

### PR DESCRIPTION
**Pull request by expenses**
See: https://github.com/paritytech/substrate/pull/6557.

### CODE REVIEW

File: node/service/src/lib.rs [hunk 1]
```diff
@@ -294,399 +311,391 @@ fn real_overseer<S: SpawnNamed>(
...
 
-		(task_manager, client)
-	}}
+	Ok((task_manager, client))
 }
 
 pub struct FullNodeHandles;
 
 /// Builds a new service for a light client.
+// TODO: this should really be a function, but the generics are tricky.
```
Comment (seunlanlege) [inline node/service/src/lib.rs:502]: don't let rob see this

Comment (seunlanlege) [reply node/service/src/lib.rs:502]: :sweat_smile:

Comment (rphmeier) [reply node/service/src/lib.rs:502]: yeah, can't commit the TODO

Comment (cecton) [reply node/service/src/lib.rs:502]: busted

File: node/service/src/lib.rs [hunk 2]
```diff
@@ -294,417 +323,412 @@ fn real_overseer<S: SpawnNamed>(
...
-					parent_hash,
-					number,
-				})
+
+fn new_full<RuntimeApi, Executor, Extrinsic>(
+	config: Configuration,
+	collating_for: Option<(CollatorId, ParaId)>,
+	_max_block_data_size: Option<u64>,
+	_authority_discovery_disabled: bool,
+	_fslot_duration: u64,
```
Comment (ordian) [inline node/service/src/lib.rs:332]: I haven't been following all the updates, but what are these unused params for? can we just remove them?

Comment (expenses) [reply node/service/src/lib.rs]: In the legacy service, `slot_duration` is used in `consensus::ProposerFactory::new`, so I think we can assume that they will be used at some point.

**Comment by rphmeier** [comment]
(won't be reviewing this one)

**Pull request merged**